edgelet/edgelet-http/src/lib.rs

```diff
@@ -217,9 +223,23 @@ where
 
         let protocol = Arc::new(protocol);
 
+        let lock = Arc::new(AtomicU32::new(0));
+
         let srv = incoming.for_each(move |(socket, addr)| {
-            let protocol = protocol.clone();
+            let lock_val = lock.fetch_add(1, Ordering::Relaxed);
```

Comment:
`fetch_*` should use `Ordering::AcqRel`, not `Relaxed`. `Relaxed` allows the update to move such that the piece of code between the two `fetch_`s is no longer between them.

Reply:
You're right, fixed.